sdk/include/opentelemetry/sdk/common/attribute_utils.h

```diff
@@ -215,6 +218,248 @@ class OrderedAttributeMap : public std::map<std::string, OwnedAttributeValue>
   AttributeConverter converter_;
 };
 
+/**
+ * Class for storing attributes.
+ */
+struct MixedAttributeMapStorage
+{
+  std::unordered_map<std::string, opentelemetry::common::AttributeValue> attributes;
+  AttributeMap owned_attributes;
+  std::unordered_map<std::string, OwnedAttributeView> owened_attributes_view;
+};
+
+/**
+ * Set an owned copy (OwnedAttributeValue) and attribute view of a non-owning AttributeValue.
+ */
+class MixedAttributeViewSetter
```

Comment:
@owent - Do we still need this class?

Reply:
It's still used by `MixedAttributeMap` , which is used by `LogRecordData`.

Reply:
@owent  As we agreed [here](https://github.com/open-telemetry/opentelemetry-cpp/pull/3147#issuecomment-2492705149), can we implement the LogRecordData to only contain the owning types, not the mixed types. The motive is to keep the public API simple and minimal. And this will be consistent with SpanData implementation.

Reply:
I think there are some differents between `LogRecordData` and `SpanData`.

`SpanData::GetAttributes` returns `std::unordered_map<std::string, opentelemetry::sdk::common::OwnedAttributeValue>` but `LogRecordData::GetAttributes` returns `std::unordered_map<std::string, opentelemetry::common::AttributeValue>`.
We need storage to store `span<T>` in `opentelemetry::common::AttributeValue`, or do you think we can break the API of 
`LogRecordData::GetAttributes` ?